src/exportComments.ts

```diff
@@ -0,0 +1,148 @@
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+
+const argv = yargs(hideBin(process.argv))
+  .option("token", {
+    alias: "t",
+    type: "string",
+    description: "GitHub personal access token",
+    demandOption: true,
+    default: process.env.GITHUB_TOKEN,
+  })
+  .option("owner", {
+    alias: "o",
+    type: "string",
+    description: "Repository owner",
+    demandOption: true,
+  })
+  .option("repo", {
+    alias: "r",
+    type: "string",
+    description: "Repository name",
+    demandOption: true,
+  })
+  .option("prs", {
+    alias: "p",
+    type: "string",
+    description: "Comma-separated list of pull request numbers",
+    demandOption: true,
+  })
+  .option("author", {
+    alias: "a",
+    type: "string",
+    description: "Author of the comments to filter with",
+    demandOption: false,
+  })
+  .parseSync(); // Use parseSync to ensure argv is not a Promise
+
+interface Comment {
+  author: string;
+  prNumber: string;
+  category: string[];
+  comment: string;
+  commentLink: string;
+}
+
+const csvWriter = createObjectCsvWriter({
+  path: "pr_comments.csv",
+  header: [
+    { id: "author", title: "Author" },
+    { id: "prNumber", title: "PR Number" },
+    { id: "category", title: "Category" },
+    { id: "comment", title: "Comment" },
+    { id: "commentLink", title: "Comment Link" },
+  ],
+});
+
+const reactionToCategory: { [key: string]: string } = {
+  "+1": "Useful",
+  eyes: "Noisy",
+  confused: "Hallucination",
+  rocket: "Teachable",
+  "-1": "Incorrect",
+};
+
+function extractCategories(reactions: any): string[] {
+  return Object.keys(reactions)
+    .filter(
+      (reaction) => reactionToCategory[reaction] && reactions[reaction] > 0
+    )
+    .map((reaction) => reactionToCategory[reaction]);
+}
+
+async function fetchComments(): Promise<void> {
+  const prNumbers = argv.prs
+    .split(",")
+    .map((pr: string) => parseInt(pr.trim(), 10));
+  let allComments: Comment[] = [];
+
+  for (const prNumber of prNumbers) {
+    const comments = await fetchCommentsForPR(prNumber);
+    allComments = allComments.concat(comments);
+  }
+
+  await csvWriter.writeRecords(allComments);
+  console.log("CSV file written successfully");
+}
+
+async function fetchCommentsForPR(prNumber: number): Promise<Comment[]> {
+  try {
+    const response = await axios.get(
+      `https://api.github.com/repos/${argv.owner}/${argv.repo}/pulls/${prNumber}/comments`,
+      {
+        headers: {
+          Authorization: `token ${argv.token}`,
+        },
+      }
+    );
+
+    let comments: Comment[] = await Promise.all(
+      response.data.map(async (comment: any) => {
+        const categories = extractCategories(comment.reactions);
+        return {
+          author: comment.user.login,
+          prNumber: prNumber.toString(),
+          category: categories,
+          comment: comment.body,
+          commentLink: comment.html_url,
+        };
+      })
+    );
+
+    if (argv.author) {
```

Comment:
The filtering needs to happen after we retrieved the comments data, so that recommendation cannot apply. Not a bad comment but incorrect in this case.